language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/util/documentsymbol/DocumentSymbolResolver.java

```diff
@@ -151,20 +182,184 @@
+        boolean isDeprecated = metadata.isPresent() &&
+                DocumentSymbolUtil.isDeprecated(metadata.get());
+        return Optional.ofNullable(createDocumentSymbol(name, symbolKind, null,
+                range, range, isDeprecated, Collections.emptyList(), this.context));
+    }
+
+    @Override
+    public Optional<DocumentSymbol> transform(AnnotationDeclarationNode annotationDeclarationNode) {
+        String name = annotationDeclarationNode.annotationTag().text();
+        SymbolKind symbolKind = SymbolKind.Property;
+        Range range = DocumentSymbolUtil.generateNodeRange(annotationDeclarationNode);
+        Optional<MetadataNode> metadata = annotationDeclarationNode.metadata();
+        boolean isDeprecated = metadata.isPresent() &&
+                DocumentSymbolUtil.isDeprecated(metadata.get());
+        return Optional.ofNullable(createDocumentSymbol(name, symbolKind, null,
+                range, range, isDeprecated, Collections.emptyList(), this.context));
+    }
+
+    @Override
+    public Optional<DocumentSymbol> transform(ObjectFieldNode objectFieldNode) {
+        String name = objectFieldNode.fieldName().text();
+        SymbolKind symbolKind = SymbolKind.Field;
+        Range range = DocumentSymbolUtil.generateNodeRange(objectFieldNode);
+        Optional<MetadataNode> metadata = objectFieldNode.metadata();
+        boolean isDeprecated = metadata.isPresent() &&
+                DocumentSymbolUtil.isDeprecated(metadata.get());
+        return Optional.ofNullable(createDocumentSymbol(name, symbolKind, null,
+                range, range, isDeprecated, Collections.emptyList(), this.context));
+    }
+
+    @Override
+    public Optional<DocumentSymbol> transform(RecordFieldNode recordFieldNode) {
+        String name = recordFieldNode.fieldName().text();
+        SymbolKind symbolKind = SymbolKind.Field;
+        Range range = DocumentSymbolUtil.generateNodeRange(recordFieldNode);
+        Optional<MetadataNode> metadata = recordFieldNode.metadata();
+        boolean isDeprecated = metadata.isPresent() &&
+                DocumentSymbolUtil.isDeprecated(metadata.get());
+        return Optional.ofNullable(createDocumentSymbol(name, symbolKind, null,
+                range, range, isDeprecated, Collections.emptyList(), this.context));
+    }
+
+    @Override
+    public Optional<DocumentSymbol> transform(RecordFieldWithDefaultValueNode recordFieldWithDefaultValueNode) {
+        String name = recordFieldWithDefaultValueNode.fieldName().text();
+        SymbolKind symbolKind = SymbolKind.Field;
+        Range range = DocumentSymbolUtil.generateNodeRange(recordFieldWithDefaultValueNode);
+        Optional<MetadataNode> metadata = recordFieldWithDefaultValueNode.metadata();
+        boolean isDeprecated = metadata.isPresent() &&
+                DocumentSymbolUtil.isDeprecated(metadata.get());
+        return Optional.ofNullable(createDocumentSymbol(name, symbolKind, null,
```

Comment:
There are cases where we pass `null` and `false` as arguments. We can provide overloaded methods to avoid this repetition.

Reply:
Done